packages/blade/src/utils/useBreakpoint/useBreakpoint.web.test.ts

```diff
@@ -0,0 +1,48 @@
+import { renderHook, act } from '@testing-library/react-hooks';
+import breakpoints from '../../tokens/global/breakpoints';
+import setupMatchMediaMock from '../mocks/setupMatchMediaMock';
+import useBreakpoint from './';
+
+describe('useBreakpoint', () => {
+  it('should detect mobile and return smallest breakpoint of mobile devices', () => {
+    setupMatchMediaMock({
+      matches: true,
+    });
+
+    const { result } = renderHook(() => useBreakpoint({ breakpoints }));
+
+    expect(result.current.matchedBreakpoint).toBe('xs');
+    expect(result.current.matchedDeviceType).toBe('mobile');
+  });
+
+  it('should match a specific media query and return the device type accordingly', () => {
+    const addEventListenerMock = jest
+      .fn()
+      .mockImplementation((_, callback: (arg: { media: string }) => void) => {
+        callback({ media: 'screen and (min-width: 769px) and (max-width: 1024px)' });
+      });
+    setupMatchMediaMock({
+      // return matches true for initial state because event object is set only once useEffect runs
+      matches: false,
+      addEventListener: addEventListenerMock,
+    });
+
+    const { result } = renderHook(() => useBreakpoint({ breakpoints }));
+
+    expect(result.current.matchedBreakpoint).toBe('l');
+    expect(result.current.matchedDeviceType).toBe('desktop');
+
+    // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+    const addEventListenerMockCallback = addEventListenerMock.mock.calls[0][1] as (arg: {
```

Comment:
@anuraghazra if you can help here 😭

Reply:
@kamleshchandnani this will fix it. 

```ts
    const addEventListenerMock = jest
      .fn<void, ((arg: { media: string; }) => void)[]>()
      .mockImplementation((_, callback) => {
        callback({ media: 'screen and (min-width: 769px) and (max-width: 1024px)' });
      });
```

Reply:
the moment I do that ts-eslint throws warning `void is only valid as a return type or generic type variable` 😭